jenkins/opensearch-dashboards/integ-test.jenkinsfile

```diff
@@ -98,29 +98,38 @@ pipeline {
                         currentBuild.result = 'ABORTED'
                         error("OSD Version $version does not match OS Version $versionOpenSearch")
                     }
-
                 }
             }
-        }
-        stage('integ-test') {
-            agent {
-                docker {
-                    label AGENT_LABEL
-                    image docker_images["$distribution"]
-                    args docker_args["$distribution"]
-                    registryUrl 'https://public.ecr.aws/'
-                    alwaysPull true
+            post {
+                always {
+                    postCleanup()
                 }
             }
+        }
+        stage('integ-test') {
+            // Required running on agent directly here to trigger docker stages in agent node, not trigger docker within docker container
+            // Can only be run in runner that is at least 50GB per container
+            agent { label AGENT_LABEL }
             steps {
                 script {
-                    def buildManifestObj = downloadBuildManifest(
+
+                    downloadBuildManifest(
                         url: BUILD_MANIFEST_URL,
                         path: BUILD_MANIFEST
                     )
 
+                    def buildManifestObj = lib.jenkins.BuildManifest.new(readYaml(file: BUILD_MANIFEST))
+                    def componentDefaultList = buildManifestObj.getNames()
+                    def componentList = COMPONENT_NAME ? COMPONENT_NAME.trim().split(" ") as List : componentDefaultList
                     String switch_user_non_root = (distribution.equals('rpm') || distribution.equals('deb')) ? 'true' : 'false'
                     echo "switch_user_non_root: ${switch_user_non_root}"
+                    echo "componentList: ${componentList}"
+
+                    for (component_check in componentList) {
+                        if (! componentDefaultList.contains(component_check)) {
```

Comment:
```suggestion
                        if (!componentDefaultList.contains(component_check)) {
```

Reply:
Add the same comment change on OS as well. Thanks.